### ISSUE TITLE
Hide server IP from resources' self href

### ISSUE THREAD

**Issue by letientai299** [spring-projects/spring-hateoas#616]
For security reason, we want no one be able to know the server IP address, but our trustest client app.
Right now, json response contains:

```
"href": "http://localhost:8080/books/1"
```

Anyway to remove the IP part from the link, so that json response is:
```
"href": "/books/1"
```

**Comment by gregturn** [comment]
A flag to only use relative links? Not so hard given we support falling back to relative URIs. 

Not sure how to flip that on for a given request especially tied to something nebulous like "truest" clients. What does that even mean?

Given this is a response to a request are you afraid of intermediate interception? I wonder if your app needs to implement a custom filter that can revise the hypermedia on the way out based on context.

**Comment by letientai299** [comment]
> "truest" clients
Sorry for the confusion. I mean trusted. 

I want to return relative links for all requests, globally, not only a single request. If there's already a "flag" within Spring HATEOS, please give me a hint :).

Thanks.

**Comment by gregturn** [comment]
Ability? Just recently. But it's a fallback not a flag (yet).

**Comment by punkstarman** [comment]
@gregturn, are you referring to #592?

**Comment by gregturn** [comment]
Yes. And it's merged to master. But that code path is only used during an exception.
